src/axum.rs

```diff
@@ -15,16 +21,35 @@ impl IntoResponse for Problem {
 
             response
                 .headers_mut()
-                .insert("content-type", "application/problem+json".parse().unwrap());
+                .insert(CONTENT_TYPE, "application/problem+json".parse().unwrap());
             response
         }
     }
 }
 
+/// Create a panic [`tower_http::catch_panic:CatchPanicLayer`] which transforms panics into RFC-7807-compatible responses.
+pub fn make_panic_handler() -> CatchPanicLayer<fn(Box<(dyn Any + Send + 'static)>) -> Response<Body>>
+{
+    CatchPanicLayer::custom(|err: Box<dyn Any + Send + 'static>| {
+        let details = if let Some(s) = err.downcast_ref::<String>() {
+            s.clone()
+        } else if let Some(s) = err.downcast_ref::<&str>() {
+            s.to_string()
+        } else {
+            "Internal server error".to_string()
+        };
+
+        crate::new(StatusCode::INTERNAL_SERVER_ERROR)
```

Comment:
I'm not sure if we want this, or maybe a builder which would have reasonable defaults, but would also allow the user to customize this response. What do you think @sazzer?

I suspect granting the user the possibility to provide their own `Problem` if they want to, leaving only the `detail` to be filled by the panic handler would be beneficial, but would also complicate the code slightly.

Reply:
I think this is reasonable. If anyone wants to customize the response we can always have an alternative with the problem builder, but for now keeping it simple makes it more usable. Especially since this is to cover a case that shouldn't ever happen!

Reply:
Have a look at the builder version. I managed to make something simple yet customizable. In my opinion it's better.

Reply:
You're right - that is pretty simple, which is the main concern for people who might want to use this with a minimum of effort...

The one thing I'm wondering about is the `detail` value. Mostly in terms of actually returning the value to the client if an appropriate type is returned - since these are potentially leaking internal details. Does the panic message still get output to logs/etc with this handler in place?

Reply:
From what I noticed the panic is returned in the console, so it's probably logged but that should be tested (I'll check it and share the results).
I wanted to make it so that the backtrace is also part of the response (if someone wants it or maybe compiles in debug mode) but from what I noticed you can't really get the backtrace easily (maybe I'm mistaken though, I'll ask on the `tower-http` project forum).
And yes you're right, having the panic message in the detail may not be a good idea for all cases. I'll make it to be the default, but also configurable, so the user may choose what they prefer.

Reply:
What if we enabled it by default iff debug assertions are enabled? That way it's useful in debug mode, but more secure by default in production mode.

Reply:
Yep, that's the plan ;).

Reply:
Okay, so I added a new `with_fill_details` method, to set whether the `Problem` `detail` field will be replaced with the panic message.

Unfortunately there's no reasonable, non-intrusive way to get the panic backtrace. There's one [very intrusive one](https://stackoverflow.com/questions/68556819/transform-backtrace-to-string-during-catch-unwind), which we certainly don't want.

[My question](https://github.com/tower-rs/tower-http/discussions/451) on the `tower-http`  forum remains unanswered, but if there's some breakthrough we can always update the implementation.

@sazzer feel free to do a review, and sorry for the long wait - I hoped to get an answer on the forum after the holidays, but no luck.

Reply:
No worries - better to take longer and get the best result :)

Could you possibly update the example to use this as well? I've got a suspicion that it won't work since the `axum` module isn't published :\

Reply:
Added the handler to the axum example and made the builder public.